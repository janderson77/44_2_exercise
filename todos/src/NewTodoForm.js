import React, {useState} from 'react'
import {v4 as uuid} from 'uuid'

const NewTodoForm = ({addTodo}) => {
    const [formData, setFormData] = useState({
        todoText: ''
    })

    const handleChange = (e) => {
        const {name, value} = e.target;
        setFormData(formData => ({
            ...formData,
            [name]: value
        }))
    }
    
    const addNewTodo = (e) => {
        e.preventDefault();
        addTodo({...formData, id: uuid()});
        setFormData({
            todoText: ''
        })
    }
    
    return(
        <div>
                <form onSubmit={addNewTodo}>
                    <div className="form-group">
                        <label>Todo Text</label>
                        <input 
                            name="todoText"
                            className="form-control"
                            value={formData.todoText}
                            onChange={handleChange}
                        />
                    </div>
                    <button
                        type="submit"
                        className="btn btn-primary"
                        onSubmit={addNewTodo}
                    >Submit</button>
                </form>
            </div>
    )
}



export default NewTodoForm