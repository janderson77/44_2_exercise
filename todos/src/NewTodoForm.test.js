import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import NewTodoForm from './NewTodoForm'

let todoForm;

describe('NewTodoForm basics', () => {
    beforeEach(() => {
        todoForm = render(<NewTodoForm/>)
    })

    it('renders without crashing', () => {
        expect(todoForm).toBeTruthy()
    })
    it('matches snapshot', () => {
        expect(todoForm.asFragment()).toMatchSnapshot()
    })
})

it('updates state with todoText input', () => {
    const mock = jest.fn()
    const {getByText} = render(<NewTodoForm addTodo={mock} />)

    fireEvent.click(getByText("Submit"))

    expect(mock).toHaveBeenCalled()
})