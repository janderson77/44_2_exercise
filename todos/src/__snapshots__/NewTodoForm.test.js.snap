// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTodoForm basics matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <div
        class="form-group"
      >
        <label>
          Todo Text
        </label>
        <input
          class="form-control"
          name="todoText"
          value=""
        />
      </div>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
